vector/src/main/java/im/vector/app/features/home/HomeDetailViewModel.kt

```diff
@@ -124,6 +137,120 @@ class HomeDetailViewModel @AssistedInject constructor(@Assisted initialState: Ho
         }
     }
 
+    private fun handleIndividualInviteByEmail(action: HomeDetailAction.InviteByEmail) {
+        val existingRoom = session.getExistingDirectRoomWithUser(action.email)
+
+        setState {
+            copy(
+                    inviteEmail = action.email,
+                    existingRoom = existingRoom
+            )
+        }
+
+        viewModelScope.launch(Dispatchers.IO) {
+            // Start the invite process by checking whether a Tchap account has been created for this email.
+            val data = tryOrNull { session.identityService().lookUp(listOf(ThreePid.Email(action.email))) }
+
+            if (data.isNullOrEmpty()) {
+                _viewEvents.post(TchapHomeViewEvents.GetPlatform(action.email))
+            } else {
+                _viewEvents.post(TchapHomeViewEvents.InviteIgnoredForDiscoveredUser(action.email))
+            }
+        }
+    }
+
+    private fun handleUnauthorizeEmail() = withState {
+        if (it.inviteEmail != null) {
```

Comment:
we can remove one imbrication level by returning from the function (it.inviteEmail ?: return@withState)